seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/sink/HiveSinkFactory.java

```diff
@@ -17,28 +17,100 @@
 
 package org.apache.seatunnel.connectors.seatunnel.hive.sink;
 
+import org.apache.seatunnel.api.configuration.ReadonlyConfig;
 import org.apache.seatunnel.api.configuration.util.OptionRule;
+import org.apache.seatunnel.api.sink.SinkReplaceNameConstant;
+import org.apache.seatunnel.api.sink.SupportMultiTableSink;
+import org.apache.seatunnel.api.table.catalog.CatalogTable;
+import org.apache.seatunnel.api.table.catalog.TableIdentifier;
+import org.apache.seatunnel.api.table.connector.TableSink;
 import org.apache.seatunnel.api.table.factory.Factory;
 import org.apache.seatunnel.api.table.factory.TableSinkFactory;
-import org.apache.seatunnel.connectors.seatunnel.hive.config.HiveConfig;
+import org.apache.seatunnel.api.table.factory.TableSinkFactoryContext;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.connectors.seatunnel.file.config.BaseSinkConfig;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.commit.FileAggregatedCommitInfo;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.commit.FileCommitInfo;
+import org.apache.seatunnel.connectors.seatunnel.file.sink.state.FileSinkState;
+import org.apache.seatunnel.connectors.seatunnel.hive.config.HiveConstants;
 
 import com.google.auto.service.AutoService;
 
-import static org.apache.seatunnel.connectors.seatunnel.hive.config.HiveConfig.ABORT_DROP_PARTITION_METADATA;
+import java.util.HashMap;
+import java.util.Map;
 
 @AutoService(Factory.class)
-public class HiveSinkFactory implements TableSinkFactory {
-    @Override
-    public String factoryIdentifier() {
-        return "Hive";
-    }
+public class HiveSinkFactory
+        implements TableSinkFactory<
+                        SeaTunnelRow, FileSinkState, FileCommitInfo, FileAggregatedCommitInfo>,
+                SupportMultiTableSink {
 
     @Override
     public OptionRule optionRule() {
         return OptionRule.builder()
-                .required(HiveConfig.TABLE_NAME)
-                .required(HiveConfig.METASTORE_URI)
-                .optional(ABORT_DROP_PARTITION_METADATA)
+                .required(HiveSinkOptions.TABLE_NAME)
```

Comment:
If you use this rule, you lose some support for the local file parameter,e.g:field_delimiter

Reply:
Why this will lose parameter? you mean this will loss parameter when you set `field_delimiter` in config file?
AFAIK, we use ReadonlyConfig will only check the parameter will not drop parameter.

Reply:
It was my misunderstanding